siliconcompiler/client.py

```diff
@@ -34,9 +34,68 @@ def get_base_url(chip):
     return remote_protocol + remote_host
 
 
+###################################
+def __post(chip, url, post_action, success_action, error_action=None):
+    '''
+    Helper function to handle the post request
+    '''
+    redirect_url = urllib.parse.urljoin(get_base_url(chip), url)
+
+    while redirect_url:
+        resp = post_action(redirect_url)
+
+        code = resp.status_code
+        if 200 <= code and code < 300:
+            return success_action(resp)
+
+        try:
+            msg_json = resp.json()
+            if 'message' in msg_json:
+                msg = msg_json['message']
+            else:
+                msg = resp.text
+        except requests.JSONDecodeError:
+            msg = resp.text
+
+        if 300 <= code and code < 400:
+            if 'Location' in resp.headers:
+                redirect_url = resp.headers['Location']
+                continue
+
+        if error_action:
+            return error_action(code, msg)
+        else:
+            chip.error(f'Server responsed with {code}: {msg}', fatal=True)
+
+
+###################################
+def __build_post_params(chip, add_job_name=True, add_job_hash=True):
+    '''
+    Helper function to build the params for the post request
+    '''
+    # Use authentication if necessary.
+    post_params = {}
+
+    if 'jobhash' in chip.status and add_job_hash:
+        post_params['job_hash'] = chip.status['jobhash']
```

Comment:
Could `job_hash` be a kwarg with default value of `None`? Then this could just be `if job_hash:`, ditto below for the job name/ID.